### PR TITLE
fix(version): Use standard-version and conventional-commits for version and release management

### PR DESCRIPTION
**Pull request by danji90**
Updates: 
- Use standard-version package to manage versioning and changelog
- Added scripts for major, minor, patch and beta publishing in package.json

### CODE REVIEW

**Comment by danji90** [comment]
@oterral For further ole commits and releases we use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) and [standard-version](https://github.com/conventional-changelog/standard-version/)

**Review by oterral — CHANGES_REQUESTED** [review]
looks very nice.

I'm in favor on replacing npm by yarn everywhere and remove the package-lock. Ask @friedjoff and @mhaertwig their opinion on this.

File: README.md [hunk 1]
```diff
@@ -48,3 +48,24 @@ editor.addControls([draw, cad]);
...
+
+[optional footer(s)]
+```
+
+The commit contains the following structural elements, to communicate intent to the consumers of your library:
+
+1. fix: a commit of the type fix patches a bug in your codebase (this correlates with PATCH in semantic versioning).
+2. feat: a commit of the type feat introduces a new feature to the codebase (this correlates with MINOR in semantic versioning).
+3. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type.
+4. types other than fix: and feat: are allowed, for example @commitlint/config-conventional (based on the the Angular convention) recommends build:, chore:, ci:, docs:, style:, refactor:, perf:, test:, and others.
```
Comment (oterral) [inline README.md:68]: @danji90 here commitlint is mentionned

Comment (danji90) [reply README.md:68]: I already implemented it locally, but am waiting for @mhaertwig or @friedjoff to comment in the ticket about if we are ok with using yarn only

**Review by oterral — APPROVED** [review]
lgtm

File: package.json [hunk 2]
```diff
@@ -50,15 +53,18 @@
...
-    "cy:test": "npm run build && start-server-and-test start http://localhost:8080 cy:run",
+    "cy:test": "yarn build && start-server-and-test start http://localhost:8080 cy:run",
     "doc": "jsdoc -p -r -c jsdoc_conf.json src -d doc README.md",
     "format": "prettier --write 'src/**/*.js' && eslint 'src/**/*.js' --fix && stylelint 'style/**/*.css' 'src/**/*.css' 'src/**/*.scss' --fix",
     "lint": "eslint 'src/**/*.js' && stylelint 'style/**/*.css' 'src/**/*.css' 'src/**/*.scss'",
-    "publish:beta": "npm run publish:public --tag beta",
-    "publish:public": "npm version && npm run build && git push origin HEAD && npm publish",
-    "start": "npm run build && webpack-dev-server --mode development --open",
+    "publish:beta": "yarn release -- --release-as patch --prerelease beta && yarn build && git push origin HEAD && yarn publish --tag beta",
+    "publish:major": "yarn release && yarn build && git push origin HEAD && yarn publish",
```
Comment (friedjoff) [inline package.json:61]: Do we really need the `--release-as` flag and NPM scripts? As far as I understand this should be handled by `standard-version` and only used in special situations. This also makes beta releases limited to patch releases.

**Review by friedjoff — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (danji90) [reply package.json]: You're right, I didn't realise it detects the version hierarchy using the commit type. Have updated this in the meantime

**Review by friedjoff — APPROVED** [review]

**Pull request merged**